terraform/aws/modules/network/main.tf

```diff
@@ -157,24 +157,135 @@ resource "aws_lb" "km_lb" {
   name            = "km-lb-${var.environment}"
   subnets         = aws_subnet.km_public_subnet.*.id
   security_groups = [aws_security_group.km_alb_sg.id]
+
+  access_logs {
+    bucket  = "<s3_bucket_name>"
+    enabled = true
+  }
 }
 
 resource "aws_lb_target_group" "km_lb_target" {
   name        = "km-lb-target-group-${var.environment}"
   port        = 80
-  protocol    = "HTTP"
+  protocol    = "HTTPS"
   vpc_id      = aws_vpc.km_vpc.id
   target_type = "ip"
-  depends_on = [ aws_lb.km_lb ]
+  depends_on  = [aws_lb.km_lb]
 }
 
```

Comment:
**Tenable.cs remediation**
 
**[HIGH]: Ensure one HTTPS listener is configured for AWS Load balancer through AWS LB Listener**
 
**Vulnerability**
Not configuring AWS Load balancer to have one HTTPS listener could impact the confidentiality of data in transit.
 
**Remediation**
Elastic Load Balancing uses Secure Sockets Layer (SSL) negotiation configurations, known as security policies, to negotiate connections between the clients and the load balancer. When you use HTTPS/SSL for your front-end connections, you can use either a predefined security policy or a custom security policy. You must deploy an SSL certificate on your load balancer.

source: tenablecs-